cpu/atmega2560/periph/gpio.c

```diff
@@ -53,23 +162,137 @@ int gpio_init_int(gpio_t dev, gpio_pp_t pushpull, gpio_flank_t flank, gpio_cb_t
 
 int gpio_read(gpio_t dev)
 {
-    return -1;
+    uint8_t pin;
+    pin = gpio_pin_map[dev - 2];
+    switch(dev) {
+      case 2:
+        DDRE  |= (1 << pin);
+        break;
+      case 3:
+        DDRE  |= (1 << pin);
+        break;
+      case 4:
+        DDRG  |= (1 << pin);
+        break;
+      case 5:
+        DDRE  |= (1 << pin);
+        break;
+      case 6:
+        DDRH  |= (1 << pin);
+        break;
+      case 7:
+        DDRH  |= (1 << pin);
+        break;
+      case 8:
+        DDRH  |= (1 << pin);
+        break;
+      case 9:
+        DDRH  |= (1 << pin);
+        break;
+      case 10:
+        DDRB  |= (1 << pin);
+        break;
+      case 11:
+        DDRB  |= (1 << pin);
+        break;
+      default:
+        return -1;
+    };
+    return 0; /* Everything alright */
```

Comment:
Looking at the gpio interface in drivers/include/periph/gpio.h the `gpio_read()` function should return a `0` when the input pin is low and a `>0` when the pin is high